### ISSUE TITLE
Race condition with CLI spinner

### ISSUE THREAD

**Issue by rossjones** [bacalhau-project/bacalhau#2229]
When running some commands, such as the non-working `bacalhau docker run ubuntu hello`, the spinner will show:

```shell
Finding node(s) for the job  ................ err  ❌  02m01.4s

Error while executing the job.
Error submitting job: Could not start spinner.: spinner already running or shutting down
```

This appears to be caused by the spinner being restarted when it is still shutting down from the previous use. 
We might simplify the `WaitForJobAndPrintResultsToUser` function (https://github.com/bacalhau-project/bacalhau/blob/46ac14168d6887c701234a25ec7126bc38573787/cmd/bacalhau/utils.go#L586)  to make it a little simpler and use the version of the spinner in spinner.go.

**Comment by wesfloyd** [comment]
Do you know if there is any temporary work around for this bug or which types of strings trigger this condition? Ie I've tried removing and re-installing the bacalhau cli, but it doesn't seem to help.

Testing with this command:
`bacalhau docker run -v QmTYadG6K4LocouLu6wFzeMXNB5z8ikdpwuGqpdZNxp5qR:/project/inputs/LC08_L1TP_001028_20220615_20220627_02_T1_mosaic.tif -o outputs:/project/outputs/ ${USERNAME}/segmentation_testbed:latest`

**Comment by rossjones** [comment]
Using ` --id-only` should help, but you'll have to rely on `describe` to check in on it.

```shell
bacalhau docker run --id-only -v QmTYadG6K4LocouLu6wFzeMXNB5z8ikdpwuGqpdZNxp5qR:/project/inputs/LC08_L1TP_001028_20220615_20220627_02_T1_mosaic.tif -o outputs:/project/outputs/ ${USERNAME}/segmentation_testbed:latest
```

**Comment by rossjones** [comment]
Was fixed in #2236
